x-pack/auditbeat/processors/add_session_metadata/provider/ebpf_provider/ebpf_provider.go

```diff
@@ -0,0 +1,170 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux
+
+package ebpf_provider
+
+import (
+	"context"
+
+	"github.com/mohae/deepcopy"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/add_session_metadata/processdb"
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/add_session_metadata/provider"
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/add_session_metadata/types"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+
+	"github.com/elastic/ebpfevents"
+)
+
+type prvdr struct {
+	ctx    context.Context
+	logger *logp.Logger
+	db     *processdb.DB
+}
+
+func NewProvider(ctx context.Context, logger *logp.Logger, db *processdb.DB) (provider.Provider, error) {
+	p := prvdr{
+		ctx:    ctx,
+		logger: logger,
+		db:     db,
+	}
+
+	l, err := ebpfevents.NewLoader()
```

Comment:
this needs to be accessed via a singleton: https://github.com/elastic/beats/blob/92941990aada25fcc921ed51834f8bcea79cfa04/auditbeat/internal/ebpf/watcher_linux.go

Reply:
Done, I've added the watcher to my branch and switched to use it